opsdroid/database/matrix/__init__.py

```diff
@@ -0,0 +1,250 @@
+        )
+        if isinstance(ori_data, RoomGetStateEventError):
+            raise RuntimeError(
+                f"Error getting {key} from matrix room {self.room_id}: {ori_data.message}({ori_data.status_code})"
+            )
+        elif ori_data.transport_response.status == 404:
+            _LOGGER.error(
+                f"Error getting {key} from matrix room {self.room_id}: Event not found"
+            )
+            return
+
+        data = ori_data.content
+
+        _LOGGER.debug(f"Got {data} from state request.")
+
+        if self._single_state_key:
+            if key in data:
+                data = {key: data[key]}
+            elif not get_full:
+                _LOGGER.debug(f"{key} doesn't exist in database")
+                return
+            else:
+                return data
+
+        for k, v in data.items():
+            if isinstance(v, dict) and "encrypted_val" in v:
+                resp = await self.connector.connection.room_get_event(
+                    room_id=self.room_id, event_id=v["encrypted_val"],
+                )
+                if isinstance(resp, RoomGetEventError):
+                    _LOGGER.error(
+                        f"Error decrypting {v['encrypted_val']} while getting "
+                        f"{key}: {resp.message}({resp.status_code})"
+                    )
+                    continue
+                data[k] = resp.event.source["content"][k]
+
+        if get_full:
+            return {**ori_data.content, **data}
+
+        return data[key]
+
+    async def delete(self, key):
+        """Delete a key from the database."""
+
+        if self.should_migrate:
+            await self.migrate_database()
+
+        # If the single state key flag is set then use that else use state key.
+        state_key = (
+            "" if self._single_state_key is True else self._single_state_key or key
+        )
+
+        data = await self.connector.connection.room_get_state_event(
+            room_id=self.room_id, event_type=self._event_type, state_key=state_key,
+        )
+        if isinstance(data, RoomGetStateEventError):
+            _LOGGER.error(
+                f"Error deleting {key} from matrix room {self.room_id}: {data.message}({data.status_code})"
+            )
+            return
+
+        if data.transport_response.status == 404:
+            _LOGGER.error(f"State event with state key '{state_key}' doesn't exist")
+            return
+        data = data.content
+
+        _LOGGER.debug(f"Got {data} from state request.")
+
+        if not isinstance(key, list):
+            key = [key]
+
+        return_value = []
+        for k in key:  # key can be a list of keys to delete
+            try:
+                return_value.append(data[k])
+                _LOGGER.debug(f"Deleting {k} from {self.room_id}")
+                del data[k]
+            except KeyError:
+                _LOGGER.debug(f"Not deleting {k}, as it doesn't exist")
```

Comment:
I think this should be logged at a higher level than debug?
```suggestion
                _LOGGER.warning(f"Unable to delete '{k}' from database in room {self.room_id} as it doesn't exist.")
```